apps/src/StudioApp.js

```diff
@@ -3191,12 +3191,20 @@ StudioApp.prototype.isResponsiveFromConfig = function(config) {
 /**
  * Checks if the level a teacher is viewing of a students has
  * not been started.
+ * For contained levels don't show the banner ever.
+ * Otherwise if its a channel backed level check for the channel. Lastly
+ * if its not a channel backed level and its not free play we know it has
+ * not been started if no progress has been made.
  */
 StudioApp.prototype.isNotStartedLevel = function(config) {
   const progress = getStore().getState().progress;
 
-  if (
-    ['Gamelab', 'Applab', 'Weblab', 'Spritelab'].includes(config.levelGameName)
+  if (config.hasContainedLevels) {
+    return false;
+  } else if (
+    ['Gamelab', 'Applab', 'Weblab', 'Spritelab', 'Dance'].includes(
```

Comment:
Sorry for the late comment, but I think we would also need some logic for artist and playlab levels which are project-backed, for example. These levels have a `project_template_level_name`. I checked on https://studio.code.org/s/express-2019/stage/7/puzzle/4 and I do not see the warning message regardless of whether the student has progress. It works fine on non-project-backed artist levels though. Opening as https://codedotorg.atlassian.net/browse/LP-983

Reply:
Oh nice find Dave! Thanks! I'm going to put that in next sprint as a follow-up.